hibernate-core/src/main/java/org/hibernate/query/internal/FetchBuilderJpa.java

```diff
@@ -6,31 +6,46 @@
  */
 package org.hibernate.query.internal;
 
+import java.util.List;
 import java.util.function.BiFunction;
 
 import org.hibernate.LockMode;
 import org.hibernate.engine.FetchTiming;
-import org.hibernate.metamodel.mapping.ModelPart;
 import org.hibernate.query.NavigablePath;
+import org.hibernate.query.results.DomainResultCreationStateImpl;
 import org.hibernate.query.results.FetchBuilder;
+import org.hibernate.query.results.MissingSqlSelectionException;
+import org.hibernate.query.results.PositionalSelectionsNotAllowedException;
+import org.hibernate.query.results.SqlSelectionImpl;
 import org.hibernate.query.results.dynamic.DynamicFetchBuilderLegacy;
+import org.hibernate.sql.ast.tree.from.TableGroup;
+import org.hibernate.sql.ast.tree.from.TableReference;
 import org.hibernate.sql.results.graph.DomainResultCreationState;
 import org.hibernate.sql.results.graph.Fetch;
 import org.hibernate.sql.results.graph.FetchParent;
 import org.hibernate.sql.results.graph.Fetchable;
 import org.hibernate.sql.results.graph.FetchableContainer;
 import org.hibernate.sql.results.jdbc.spi.JdbcValuesMetadata;
 
+import static org.hibernate.query.results.ResultsHelper.impl;
+import static org.hibernate.query.results.ResultsHelper.jdbcPositionToValuesArrayPosition;
+import static org.hibernate.sql.ast.spi.SqlExpressionResolver.createColumnReferenceKey;
+
 /**
  * @author Steve Ebersole
  */
 public class FetchBuilderJpa implements FetchBuilder {
 	private final NavigablePath navigablePath;
 	private final String attributePath;
+	private final List<String> columnNames;
 
-	public FetchBuilderJpa(NavigablePath navigablePath, String attributePath) {
+	public FetchBuilderJpa(
+			NavigablePath navigablePath,
+			String attributePath,
+			List<String> columnNames) {
 		this.navigablePath = navigablePath;
 		this.attributePath = attributePath;
+		this.columnNames = columnNames;
```

Comment:
Are these passed `columns` ever used here?

Reply:
Should I delete `FetchBuilderJpa` and `FetchMementoJpa`? They were only used by this code section https://github.com/hibernate/hibernate-orm/pull/3662/files#diff-ef7d256eac99c70fe4b496c2b04708b3dc6f71a99d333ac1aeb18c344ef16484L398 which I changed